learning/dashboard/webserver/readme.md

```diff
@@ -0,0 +1,161 @@
+
+**Jigsaw :**
+
+The World Wide Web Consortium (W3C) created the web server Jigsaw as a reference implementation of the HTTP/1.1 protocol. For programmers curious about HTTP protocols and web server technology, Jigsaw is still a popular option.
+Jigsaw's modular design, which enables customers to easily adapt and increase the server's capability, is one of its primary benefits. CGI, server-side programming languages like PHP and Python, and SSL/TLS encryption are just a few of the features it provides. Moreover, it includes built-in support for virtual hosting, which enables the hosting of numerous websites on a single server.
+
+#	How to setup and configure Nginx ?
+
+##	Nginx installation on Ubuntu 20.04
+
+A package manager is typically included with every operating system. You have a choice between using apt or apt-get with Ubuntu 20.04. As this is a new server, you first need to update the package manager.
+
+Step 1 :  Upgrade the package management to the most recent version that is compatible:
+You can either use the apt or apt-get package manager to install Nginx.
+sudo apt updatesudo apt-get update
+Now, you are ready to install Ngnix. When you are logged in with a non-root account with sudo rights, you will need to prefix every task that requires admin privileges with sudo.
+
+Step 2 : Install Ngnix web server using apt:	
+sudo apt install nginxsudo apt-get install nginx
+
+Although the service starts automatically after installing Nginx, before you can test whether Nginx is working correctly or not, you will need to configure the OS firewall to access Nginx.
+
+## Configure Firewall for Nginx :
+
+The most popular method for creating iptables' ingress and egress rules. Uncomplicated Firewall, or ufw, is a user-friendly wrapper for iptables that is available in Ubuntu 20.04. Although it comes installed with Ubuntu 20.04 but to make sure that it is appropriately installed, install it manually. After installing Nginx, follow the instructions below to set up the firewall
+
+Step 1: Install ufw:
+sudo apt install ufw
+
+Step 2: Check if ufw is working fine:
+sudo ufw status
+
+Step 3: Allow traffic over HTTP and recheck the status:
+sudo ufw allow ‘Nginx HTTP’
+
+Step 4: Allow traffic over TCP (OpenSSH) too if you want to log into the machine again:
+sudo ufw allow ‘OpenSSH’
+
+Step 5: If the status was inactive when you checked earlier, activate it:
+sudo ufw enable
+
+The choice of whether to implement the changes or not will be presented to you. Be aware that you won't be able to enter into the machine again if you deactivate TCP traffic or add a deny OpenSSH rule to the firewall. Let’s now cover some crucial Nginx commands for administration and management
+
+##Nginx Installation and Management :
+
+Step 1: Check the status of the Nginx server:
+               sudo systemctl status nginx
```

Comment:
Consider using a code block for commands and code snippets
`sudo systemctl status nginx`